obspy/io/sac/sactrace.py

```diff
@@ -1158,10 +1176,23 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                                             checksize=checksize)
         if not debug_strings:
             for i, val in enumerate(hs):
-                val = _ut._clean_str(val, strip_whitespace=False)
+                if encoding.lower() != 'ascii':  # throw warning
+                    msg = ("Using encoding other than ASCII: {}\n"
+                           "All non-ASCII characters will be replaced by "
+                           "?".format(encoding))
```

Comment:
Those are two different warnings. The warning here tells users that they are using encoding other than "ASCII" which is not ideal and non-ASCII character will be converted.  The warning in `_clean_str()` was to tell users that the conversion actually happened. I agree that throwing two warnings here may be a bit too verbose. I will think about how to merge them.